File: fechamento.py
def calcular_total_do_dia():
    total_do_dia = 0.0

    try:
        with open("recibo.txt", "r") as recibo_arquivo:
            for linha in recibo_arquivo:
                valor_do_recibo = 0.0
                if "Valor Total do Pedido: R$" in linha:
                    valor_do_recibo = float(linha.split("R$")[1])
                    total_do_dia += valor_do_recibo
    except FileNotFoundError:
        print("Nenhum recibo encontrado. O restaurante não fez vendas hoje.")
        return None

    return total_do_dia


def main():
    total_do_dia = calcular_total_do_dia()

    if total_do_dia is not None:
        print(f"Total do dia: R$ {total_do_dia:.2f}")


if __name__ == "__main__":
    main()

File: fechamento.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    FILE *reciboArquivo;
    double totalDoDia = 0.0;
    char linha[256];

    reciboArquivo = fopen("recibo.txt", "r");

    if (reciboArquivo == NULL) {
        printf("Nenhum recibo encontrado. O restaurante não fez vendas hoje.\n");
        return 1;
    }

    while (fgets(linha, sizeof(linha), reciboArquivo) != NULL) {
        double valorDoRecibo = 0.0;
        if (sscanf(linha, "Valor Total do Pedido: R$ %lf", &valorDoRecibo) == 1) {
            totalDoDia += valorDoRecibo;
        }
    }

    fclose(reciboArquivo);

    printf("Total do dia: R$ %.2f\n", totalDoDia);

    return 0;
}
